dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/programrule/implementers/event/RuleEngineErrorExecutor.java

```diff
@@ -25,23 +25,42 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package org.hisp.dhis.tracker.programrule;
+package org.hisp.dhis.tracker.programrule.implementers.event;
 
-import java.util.List;
+import static org.hisp.dhis.tracker.programrule.ProgramRuleIssue.warning;
+
+import java.util.Optional;
+
+import lombok.RequiredArgsConstructor;
 
-import org.hisp.dhis.rules.models.RuleEffect;
 import org.hisp.dhis.tracker.bundle.TrackerBundle;
 import org.hisp.dhis.tracker.domain.Event;
+import org.hisp.dhis.tracker.programrule.ProgramRuleIssue;
+import org.hisp.dhis.tracker.programrule.implementers.RuleActionExecutor;
+import org.hisp.dhis.tracker.validation.ValidationCode;
 
 /**
- * @author Enrico Colasante
+ * This executor log as a warning any error raised by rule engine execution
```

Comment:
The word log confuses me as I think of actually logging to stdout/disk. Maybe

```suggestion
 * This executor emits a warning for any error raised by the rule engine execution.
```

or creates, adds if emits does not feel right 🤷🏻